### PR TITLE
[doc] Fix registers in entropy_src documentation

### PR DESCRIPTION
**Pull request by jwnrt**
This PR fixes some of the names of registers in the entropy_src documentation.

The main two fixes here are:

1. `CONF.ENABLE` was moved to `MODULE_ENABLE` in 51fde5a9830ab1501b01d953c7651c63727a546d.
2. `REGEN` was renamed to `REGWEN` in ef348c9995b53217479ebda6af8f7501042ecb72.

I have also added some convenience links to registers and fields that weren't linked previously.

### CODE REVIEW

File: hw/ip/entropy_src/doc/theory_of_operation.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ As already described, this IP block will collect bits of entropy for firmware or
 This revision supports only an external interface for a PTRNG noise source implementation.
 
 The first step is initialization and enabling.
-The PTRNG noise source mode is selected when the `ENABLE` field will be set.
+The PTRNG noise source mode is selected when the [`MODULE_ENABLE`](registers.html#module_enable) field is be set.
```
Comment (rswarbrick) [inline hw/ip/entropy_src/doc/theory_of_operation.md:7]: Nit: Stray "is"

**Review by rswarbrick — APPROVED** [review]
Tiny nit, but this looks good otherwise. Thanks for sorting out the names!

File: hw/ip/entropy_src/doc/theory_of_operation.md (hunk 1 above)
Comment (jwnrt) [reply hw/ip/entropy_src/doc/theory_of_operation.md]: Oops, thanks

The `is` was intentional, but I also meant to delete the `be`

File: hw/ip/entropy_src/README.md [hunk 2]
```diff
@@ -56,7 +54,7 @@ The repetition count test fails if any sequence of bits continuously asserts the
 The thresholds for these tests should be chosen to achieve a low false-positive rate (&alpha;) given a conservative estimate of the manufacturing tolerances of the PTRNG noise source.
 The combined choice of threshold and window size then determine the false-negative rate (&beta;), or the probability of missing statistical defects at any particular magnitude.
 
-When the IP is disabled by clearing the `ENABLE` bit in [`CONF`](data/entropy_src.hjson#conf), all heath checks are disabled and all counters internal to the health checks are reset.
+When the IP is disabled by clearing the [`MODULE_ENABLE`](./doc/registers.html#MODULE_ENABLE) register, all heath checks are disabled and all counters internal to the health checks are reset.
```
Comment (vrozic) [inline hw/ip/entropy_src/README.md:57]: nit: `all heath checks` -> `all health checks`

**Review by vrozic — APPROVED** [review]

**Comment by luismarques** [comment]
LGTM.

Taking a step back, why did we not notice the broken link/anchors before? Were our current checks supposed to catch that? Were the broken ones grandfathered into an allow list or something...? @HU90m?

**Review by HU90m — APPROVED** [review]

**Comment by HU90m** [comment]
> Taking a step back, why did we not notice the broken link/anchors before? Were our current checks supposed to catch that? Were the broken ones grandfathered into an allow list or something...? @HU90m?

Adding the anchor checks was blocked on something. I can't remember what. I'll revisit it tomorrow.

**Pull request merged**